osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/services/MonitoringService.java

```diff
@@ -96,24 +96,24 @@ public ResponseData dequeueGetPowerQualityProfileDataResponseData(final String c
 
     public String enqueueClearAlarmRegisterRequestData(@Identification final String organisationIdentification,
             @Identification final String deviceIdentification, final ClearAlarmRegisterRequest requestData,
-            final int messagePriority, final Long scheduleTime) throws FunctionalException {
+            final int messagePriority, final Long scheduleTime, final boolean bypassRetry) throws FunctionalException {
 
         return this.enqueueRequestData(organisationIdentification, deviceIdentification, requestData, messagePriority,
-                scheduleTime, DeviceFunction.CLEAR_ALARM_REGISTER, MessageType.CLEAR_ALARM_REGISTER);
+                scheduleTime, DeviceFunction.CLEAR_ALARM_REGISTER, MessageType.CLEAR_ALARM_REGISTER, bypassRetry);
     }
 
     public String enqueueActualPowerQualityRequestData(@Identification final String organisationIdentification,
             @Identification final String deviceIdentification, final ActualPowerQualityRequest requestData,
-            final int messagePriority, final Long scheduleTime) throws FunctionalException {
+            final int messagePriority, final Long scheduleTime, final boolean bypassRetry) throws FunctionalException {
 
         return this.enqueueRequestData(organisationIdentification, deviceIdentification, requestData, messagePriority,
-                scheduleTime, DeviceFunction.GET_ACTUAL_POWER_QUALITY, MessageType.GET_ACTUAL_POWER_QUALITY);
+                scheduleTime, DeviceFunction.GET_ACTUAL_POWER_QUALITY, MessageType.GET_ACTUAL_POWER_QUALITY, bypassRetry);
     }
 
     private String enqueueRequestData(@Identification final String organisationIdentification,
```

Comment:
Consider splitting this method up into 3 methods for:
- checking authorizations
- creating metadata (incl. correlation uid)
- sending message

Reply:
Changed